File: backend/schemas/net_worth_projection.py
from pydantic import BaseModel


class NetWorthProjectionPoint(
    BaseModel
):
    year: int

    investment_value: float

    real_estate_value: float
    real_estate_loan_balance: float
    real_estate_equity: float

    net_worth: float


class NetWorthProjectionResponse(
    BaseModel
):
    currency: str

    monthly_contribution: float
    annual_return_percent: float
    annual_property_growth: float

    projection_years: int

    starting_investment_value: float
    starting_real_estate_equity: float
    starting_net_worth: float

    final_investment_value: float
    final_real_estate_equity: float
    final_net_worth: float

    yearly_projection: list[
        NetWorthProjectionPoint
    ]

File: frontend/src/types/netWorthProjection.ts
export interface NetWorthProjectionPoint {
  year: number

  investment_value: number

  real_estate_value: number
  real_estate_loan_balance: number
  real_estate_equity: number

  net_worth: number
}


export interface NetWorthProjection {
  currency: string

  monthly_contribution: number
  annual_return_percent: number
  annual_property_growth: number

  projection_years: number

  starting_investment_value: number
  starting_real_estate_equity: number
  starting_net_worth: number

  final_investment_value: number
  final_real_estate_equity: number
  final_net_worth: number

  yearly_projection: (
    NetWorthProjectionPoint[]
  )
}